### DISCUSSION TITLE
Optimisation slows down exponentially over time

### DISCUSSION THREAD

**Discussion by Knuttatutta** [Tunny-gh/Tunny_in_Open_Sea#151] [General]
Hi @hrntsm ,

Do you also have issues with the optimisation slowing down quite quickly until it becomes extremely and prohibitively slow? In the beginning a trial can take 0.05 seconds, and at 500 trials it can take 0.5 second..

This happens with all types of storage method (.db, .log, and in memory). In memory is of course faster at first, but it also really slows down.

I haven't looked into yet what might be causing it, but I was wondering if you were maybe aware of it and the cause?

Edit:
I'm not 100% sure, but with or without geometry attributes doesn't seem to affect the slowing down behaviour. I do think it's maybe a tiny bit slower in general with geometry, but I don't think it affects how it's slowing down. But again, not completely sure

**Comment by hrntsm** [comment]
Hi @Knuttatutta,
Thanks for continuing to raise issues. The speed of analysis is an issue that needs to be addressed as you have been in contact with me for some time, so I will work on it.

What optimisation algorithm is being used? Is this a common trend across all algorithms?
When Bayesian optimisation is used, the number of trials becomes progressively slower as the number of trials increases due to the optimisation algorithm. The order of computation is $O(n \log{n})$ for TPE and $O(n^3)$ for Botorch, where n is the number of trials.
So if you are using these algorithms, it is not surprising that they slow down at the rate you have described.
Computation times can be found in the following documents from Optuna
https://optuna.readthedocs.io/en/stable/reference/samplers/index.html

If other algorithms are used, one of the possible causes is that the storage process is gradually slowed down due to the larger file size of the optimisation results, but as this is also the case in in-memory mode, it will be investigated.

**Reply by Knuttatutta** [reply]
I'm mostly just using NSGA-II actually.

I now have a .log file that is 3.7GB in size, and when I run NSGA-II optimisation with JournalStorage now with the file being that large, it's fast in the beginning, but then slows down afterwards, so it doesn't appear to be so affected by the file size.

The file size does affect the speed of such things as opening Tunny and starting an optimisation run.

So in the end there are two operations that become slower and slower:

1. study.best_trials in RunOptimize: This is the worst one. It was taking like 30-40x longer after 3-400 trials.
2. study.trials in Compute2dHypervolumeRatio: This one is not as bad, but it was like 10x slower

**Reply by hrntsm** [reply]
Thanks for your survey.
It is true that these methods are considered to become slower and slower as the number of trials increases.
So these may be one of the causes.

These two are only used to display the current value in the UI, so I will change them so that they can be configured to be displayed or not.

**Reply by hrntsm** [reply]
I make issue about this.
https://github.com/hrntsm/Tunny/issues/152

**Comment by Knuttatutta** [comment]
Edit: Moved to reply

**Comment by Knuttatutta** [comment]
Edit: Moved to reply
